platform/ui/src/hooks/useMedia.js

```diff
@@ -1,54 +1,235 @@
+  useEffect(() => {
+    const { mediaQueryMap } = state;
+    mediaQueryMap.forEach(mql => {
+      mql.removeListener(onMediaQueryChange);
+      mql.addListener(onMediaQueryChange);
+    });
+  }, [state.mediaQueryMap]);
 
-      // Set a listener for each media query with above handler as callback.
-      mediaQueryLists.forEach(mql => mql.addListener(handler));
+  useEffect(() => {
+    mount.current = true;
 
-      // Remove listeners on cleanup
-      return () => mediaQueryLists.forEach(mql => mql.removeListener(handler));
-    },
-    [] // Empty array ensures effect is only run on mount and unmount
+    return () => {
+      const { mediaQueryMap } = state;
+      mediaQueryMap.forEach(mql => {
+        mql.removeListener(onMediaQueryChange);
+      });
+      mount.current = false;
+    };
+  }, []);
+
+  return state.displaySize;
+};
+/**
+ * Hook to get a content based on current displayMedia size.
+ *
+ * It uses useDisplayMediaSize. Hook immutable on content changing (i.e. mutable only when displayMediaSize changes)
+ *
+ * Current hook only offers content, it wont expose method to change its state.
+ *
+ * @param {Array} mediaQueriesStringList - array of string media queries to be parsed
+ * @param {Array} mediaTypesAliases - array of aliases. Each value represents one mediaQueryList from array mediaQueriesStringList
+ * @param {String} defaultMediaType - default mediaTypeAlias
+ * @param {Object} contentArrayMap - Mapping object for mediaTypesAliases to content (of any type).
+ * @param {Any} defaultContent - Default content to be used in case current displayMedia size not present on contentArrayMap
+ * @returns {Any} current content based on displayMedia size.
+ *
+ * @example <caption>Example to getContent based on displayMedia size</caption>
+ *
+ *    const currentComponent = useDisplayMediaContent(
+ *    ['(min-width: 1500px)', '(min-width: 1000px)', '(min-width: 600px)'],
+ *    // Value to return for matched media query
+ *    ['large', 'medium', 'small'],
+ *      // Default value
+ *    'medium',
+ *    {"small": ComponentA, "medium": ComponentB},
+ *    ComponentA);
+ *
+ *    const currentObject = useDisplayMediaContent(
+ *    ['(min-width: 1500px)', '(min-width: 1000px)', '(min-width: 600px)'],
+ *    // Value to return for matched media query
+ *    ['large', 'medium', 'small'],
+ *      // Default value
+ *    'medium',
+ *    {"small": ObjectA, "medium": ObjectB},
+ *    ObjectB);
+ */
+const useDisplayMediaContent = (
+  mediaQueriesStringList,
+  mediaTypesAliases,
+  defaultMediaType,
+  contentArrayMap,
+  defaultContent
+) => {
+  const displaySize = useDisplayMediaSize(
+    mediaQueriesStringList,
+    mediaTypesAliases,
+    defaultMediaType
   );
 
-  return value;
-}
+  const getContent = () => {
+    const content =
+      displaySize in contentArrayMap
+        ? contentArrayMap[displaySize]
+        : defaultContent;
+    return content;
+  };
+
+  return getContent();
+};
```

Comment:
Instead of calling `useDisplayMediaSize` internally, we should probably setup the JSDOC Example to show:

```js
const size = useMedia(
    // Media queries
    ['(min-width: 1500px)', '(min-width: 1000px)', '(min-width: 600px)'],
    // Value to return for matched media query
    ['large', 'medium', 'small'],
    // Default value
    'medium');

// Pass size to a child component, and in child component:
const content = useContent(
    props.size,
    {"small": ObjectA, "medium": ObjectB, "large": ObjectB },
    "medium");
```

Reply:
I'm not 100% sure `useContent` is necessary though. As you could just do:

```js
const contentTemplates = { /* ... */ };
const defaultTemplate = contentTemplates ["medium"];

const content = props.size ? contentTemplates[props.size] : defaultTemplate;
```